tikv-client/src/main/java/com/pingcap/tikv/types/DateTimeType.java

```diff
@@ -56,7 +57,8 @@ protected Object doConvertToTiDBType(Object value)
       throws ConvertNotSupportException {
     java.sql.Timestamp result;
     if (value instanceof Long) {
-      result = new java.sql.Timestamp((Long) value);
+      throw new ConvertNotSupportException(value.getClass().getName(), this.getClass().getName());
```

Comment:
I think this logic should be something like `someExtendedTimestamp.toTimestamp()`?